cocotb/__init__.py

```diff
@@ -135,8 +136,11 @@ def _initialise_testbench(root_name):
     The environment variable :envvar:`COCOTB_HOOKS`, if present, contains a
     comma-separated list of modules to be executed before the first test.
     """
+    _rlock = threading.RLock()
```

Comment:
This defeats the point of the rlock, which looks to have been to avoid `_initialise_testbench` being called twice concurrently. Locks can't do anything if only one function has them in scope.

Reply:
When would that ever happen? Perhaps we should just remove it.

Reply:
I'll move that change to a separate PR.